aws/resource_aws_lex_bot_test.go

```diff
@@ -0,0 +1,241 @@
+					testAccCheckAwsLexBotExists(testBotID, "$LATEST"),
+
+					// user provided attributes
+					resource.TestCheckResourceAttr(resourceName, "abort_statement.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "abort_statement.0.message.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "child_directed", "false"),
+					resource.TestCheckResourceAttr(resourceName, "clarification_prompt.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "clarification_prompt.0.max_attempts", "2"),
+					resource.TestCheckResourceAttr(resourceName, "clarification_prompt.0.message.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "description", "Bot to order flowers on the behalf of a user"),
+					resource.TestCheckResourceAttr(resourceName, "idle_session_ttl_in_seconds", "600"),
+					resource.TestCheckResourceAttr(resourceName, "intent.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "locale", lexmodelbuildingservice.LocaleEnUs),
+					resource.TestCheckResourceAttr(resourceName, "name", testBotID),
+					resource.TestCheckResourceAttr(resourceName, "process_behavior", lexmodelbuildingservice.ProcessBehaviorSave),
+					resource.TestCheckResourceAttr(resourceName, "voice_id", "Salli"),
+
+					// computed attributes
+					resource.TestCheckResourceAttrSet(resourceName, "checksum"),
+					resource.TestCheckResourceAttrSet(resourceName, "status"),
+				),
+			},
+			{
+				ResourceName: resourceName,
+				ImportState:  true,
+			},
+			{
+				Config: fmt.Sprintf(testAccAwsLexBotUpdateConfig, testID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+
+					// user provided attributes
+					resource.TestCheckResourceAttr(resourceName, "abort_statement.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "abort_statement.0.message.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "child_directed", "true"),
+					resource.TestCheckResourceAttr(resourceName, "clarification_prompt.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "clarification_prompt.0.max_attempts", "3"),
+					resource.TestCheckResourceAttr(resourceName, "clarification_prompt.0.message.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "description", "Bot to order flowers"),
+					resource.TestCheckResourceAttr(resourceName, "idle_session_ttl_in_seconds", "300"),
+					resource.TestCheckResourceAttr(resourceName, "intent.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "locale", lexmodelbuildingservice.LocaleEnUs),
+					resource.TestCheckResourceAttr(resourceName, "name", testBotID),
+					resource.TestCheckResourceAttr(resourceName, "process_behavior", lexmodelbuildingservice.ProcessBehaviorBuild),
+					resource.TestCheckResourceAttr(resourceName, "voice_id", "Ivy"),
+
+					// computed attributes
+					resource.TestCheckResourceAttrSet(resourceName, "checksum"),
+					resource.TestCheckResourceAttrSet(resourceName, "status"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
We should add the `TestAccAwsLexBot_disappears` test here, similar to Slot Type and Intent